### PR TITLE
Feat/rspec

### PR DESCRIPTION
**Pull request by adebsa2401**
In this PR,
- I set validation in `User` and `Post` model
- I set spec for model methods and validation
Cheers

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      post.title = 'Post 1'
+      post.text = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.'
+      post.author = User.create do |user|
+        user.name = 'John Doe'
+        user.photo = 'https://picsum.photos/200'
+        user.bio = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.'
+      end
+    end
+  end
+
```
Comment (BenMukebo) [inline spec/models/post_spec.rb:15]: - [ ] Great job so far 👏🏻 However, all custom methods in the models must have a corresponding unit test. Actually, you are missing a test to test the `update_posts_counter` method of your Post model, please add this ✍🏻

```ruby
  def update_posts_counter
    author.increment!(:posts_counter)
  end
```

File: spec/models/comment_spec.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
+  subject do
+    Comment.create do |comment|
+      comment.post = Post.create do |post|
+        post.title = 'Post 1'
```
Comment (BenMukebo) [inline spec/models/comment_spec.rb:7]: - [ ] The same here. So far great job 👏🏻 However,  you are missing a test to test the `update_comments_counter` method of your Comment model, please add this ✍🏻

```ruby
  def update_comments_counter
    post.increment!(:comments_counter)
  end
```

File: spec/models/like_spec.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+require 'rails_helper'
+
+RSpec.describe Like, type: :model do
+  subject do
+    Like.create do |like|
+      like.author = User.create do |user|
+        user.name = 'John Doe'
+        user.photo = 'https://picsum.photos/200'
+        user.bio = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.'
+      end
```
Comment (BenMukebo) [inline spec/models/like_spec.rb:10]: - [ ] The same here. So far great job 👏🏻 However,  you are missing a test to test the `update_likes_counter` method of your Like model, please add this ✍🏻

```ruby
  def update_likes_counter
    post.increment!(:likes_counter)
  end
```

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @adebsa2401,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- Gitflow was correctly used.  ✔️   
- Good commit messages ✔️ 
- Descriptive PR.  ✔️ 
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

- [ ] When I click the link to the `issues page` link, I get this 404 `error`.   Please, consider fixing the broken link by adding the issue link in the contributing sections: 

```
## 🤝 Contributing

Contributions, issues, and feature requests are welcome!

Feel free to check the [issues page](../../issues/).
```
- I also recommend that you delete this 👇🏻 in your License section as these are instructions on how to complete the License section.

image


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KayLemba — APPROVED** [review]
[[# STATUS: APPROVED :green_circle: :clap: 

Hello @adebsa2401  , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

HIGHLIGHTS:

- [x]  Your repo is professional 
- [x] Descriptive P.R
- [x] Proffessional README
- [x] Tests are passing 








 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]](https://github.com/adebsa2401/BlogApp.git)

**Pull request merged**